### PR TITLE
Fix/deepsets slurm out root and nounset

### PR DESCRIPTION
**Pull request by danielluna90**
Deepset example did not work initially, following the procedure listed in the README.md with a clean environment. Took some time to debug it, this is the product. Going to link this to Issue #119.

### CODE REVIEW

File: slurm/deepsets_job.slurm [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #SBATCH --output=logs/deepsets-%x-%j.out
 #SBATCH --error=logs/deepsets-%x-%j.err
 
-set -euo pipefail
+set -eo pipefail
```
Comment (danielluna90) [inline slurm/deepsets_job.slurm:10]: This seems to make micromamba happy. When starting the SLURM job it raises a variable unset error.

**Review by jpivarski — CHANGES_REQUESTED** [review]
Issue #119 is asking for the whole feature engineering. This PR seems to be just to get the minimal example running.

Within that scope, it looks okay apart from one issue:

The new PyTorch-compat fallback in [deepsets_model.py:106](/home/jpivarski/dsi/vanguard/deepsets_model.py#L106) is broken. `torch.scatter_reduce` is called as `torch.scatter_reduce(encoded, 0, idx, reduce="amax", output_size=batch_size)`, but the actual signature is `torch.scatter_reduce(input, dim, index, src, reduce, *, include_self=True)`. It requires a `src` argument and does not accept `output_size`. On a runtime that lacks `Tensor.scatter_reduce_` and needs this fallback, the model will fail at inference/training start with a `TypeError`. That means the claimed compatibility fix does not actually work for the old-PyTorch case it targets.

**Review by jpivarski — APPROVED** [review]
The problem above has been resolved (checked by inspecting the code, not attempting to run in an older PyTorch to prove compatibility).

This PR addresses a smaller problem than issue #119, so it's important that merging this does _not_ close the issue.

Auto-merge is on, so as soon as I accept the changes, it will merge. Here it goes!
